src/Twig/StimulusTwigExtension.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony WebpackEncoreBundle package.
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Twig;
+
+use Symfony\WebpackEncoreBundle\Exception\MissingPackageException;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class StimulusTwigExtension extends AbstractExtension
+{
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('stimulus_controller', [$this, 'renderStimulusController'], ['needs_environment' => true, 'is_safe' => ['all']]),
+        ];
+    }
+
+    public function renderStimulusController(Environment $env, array $data): string
+    {
+        if (!$data) {
+            return '';
+        }
+
+        $controllers = [];
+        $values = [];
+
+        foreach ($data as $controllerName => $controllerValues) {
+            $controllerName = twig_escape_filter($env, $controllerName, 'html_attr');
+            $controllers[] = $controllerName;
+
+            foreach ($controllerValues as $key => $value) {
+                if (!is_scalar($value)) {
+                    $value = json_encode($value);
+                }
+
+                $key = $this->convertKeyCase(twig_escape_filter($env, $key, 'html_attr'));
```

Comment:
escaping should happen after convertKeyCase, isn't it?

Reply:
I'm not sure :) . In practice this probably will never happen (you need to know the key, it can't really be user data), but just in case I thought if would be better to always have a formatted value for Stimulus not to break.